sky/clouds/kubernetes.py

```diff
@@ -241,6 +240,10 @@ def make_deploy_resources_variables(
             # TODO(romilb): Allow user to specify custom images
             'image_id': image_id,
         }
+
+        if resources.ports:
```

Comment:
```suggestion
        if resources.ports is not None:
```
nit: explicitly check `None`